src/ripple/app/paths/PathState.cpp

```diff
@@ -197,7 +197,7 @@ TER PathState::pushNode (
 
     node.uFlags = iType;
     node.issue_.currency = hasCurrency ?
-            currency : backNode.issue_.currency;
+            currency : backNodeOuter.issue_.currency;
```

Comment:
Comment on line 266 needs to be updated.
eg.
```// Note: backNodeOuter may no longer be the immediately previous node.```

Reply:
Fixed in 42d4b481a `Misc fixups in shadow warnings commit`